src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -413,22 +434,105 @@ def save_tile(ix, iy, das):
         # Run in a separate thread
         threading.Thread(target=save_tile, args=(ix, iy, das), ).start()
 
-    def _acquireTile(self, i, ix, iy):
+    def _constructCube(self, images):
+        """
+        Construct xyz cube from a  z stack of images
+        :param images: (list) list of z ordered images
+        :return: (DataArray) the da of the xyz cube
+        """
+        # images is a list of 3 dim data arrays.
+        ret = []
+        for image in images:
+            stack = numpy.dstack(image)
+            stack = numpy.swapaxes(stack, 1, 2)
+            ret.append(stack[0])
+
+        # Add back metadata
+        metadata3d = copy.copy(images[0].metadata)
+        # Extend pixel size to 3D
+        ps_x, ps_y = metadata3d[model.MD_PIXEL_SIZE]
+        ps_z = ZSTEP
+
+        # Computer cube centre
+        c_x, c_y = metadata3d[model.MD_POS]
+        c_z = self._zlevels[len(self._zlevels)//2]  # Assuming zlevels are ordered
+        metadata3d[model.MD_POS] = (c_x, c_y, c_z)
+
+        # For a negative pixel size, convert to a positive and flip the z axis
+        if ps_z < 0:
+            ret = numpy.flipud(ret)
+            ps_z = -ps_z
+
+        metadata3d[model.MD_PIXEL_SIZE] = (ps_x, ps_y, abs(ps_z))
+        metadata3d[model.MD_DIMS] = "ZYX"
+
+        ret = DataArray(ret, metadata3d)
+
+        return ret
+
+    def _acquireStreamCompressedZStack(self, i, ix, iy, stream):
+        """
+        Acquire a compressed zstack image for the given stream.
+        The method does the following:
+            - Move focus over the list of zlevels
+            - For each focus level acquire image of the stream
+            - Construct xyz cube for the acquired zstack
+            - Compress the cube into a single image using 'maximum intensity projection'
+        :return DataArray: Acquired da for the current tile stream
+        """
+        zstack = []
+        for z in self._zlevels:
+            logging.debug(f"Moving focus for tile {ix}x{iy} to {z}.")
+            self._focus.moveAbsSync({'z': z})
+            da = self._acquireStreamTile(i, ix, iy, stream)
+            zstack.append(da)
+
+        if self._future._task_state == CANCELLED:
+            raise CancelledError()
+        logging.debug(f"Zstack acquisition for tile {ix}x{iy}, stream {stream.name} finished, compressing data into a single image.")
+        # Convert zstack into a cube
+        fm_cube = self._constructCube(zstack)
```

Comment:
Please "log" this fm_cube to a separate file. At least at the beginning, it can be handy to check the whole zstack.